fili-core/src/main/java/com/yahoo/bard/webservice/druid/model/query/DruidAggregationQuery.java

```diff
@@ -96,4 +101,22 @@
     default DruidAggregationQuery<?> getInnermostQuery() {
         return (DruidAggregationQuery<?>) DruidFactQuery.super.getInnermostQuery();
     }
+
+    /**
+     * Produce the schema-defining columns for a given druid query.
+     *
+     * @return A stream of columns based on the signature of the Druid Query.
+     */
+    default Stream<Column> buildSchemaColumns() {
+        return Stream.of(
+                getDimensions().stream()
+                        .map(DimensionColumn::new),
+                getAggregations().stream()
+                        .map(Aggregation::getName)
+                        .map(MetricColumn::new),
+                getPostAggregations().stream()
```

Comment:
Instead of doing streams over both of these, consider adding a `getMetricFieldNames()` method that gives back aggregations _and_ post-aggregations (since they both implement `metricField`.

It would make this method a little bit simpler, and provide a useful, reusable hook for other things to get a unified view of aggs and post-aggs.